web/src/engine/predictive-text/worker-thread/src/main/correction/context-tracker.ts

```diff
@@ -42,9 +42,15 @@ export class TrackedContextToken {
   isWhitespace?: boolean;
 
   transformDistributions: Distribution<Transform>[] = [];
-  replacements: TrackedContextSuggestion[];
+  replacements: TrackedContextSuggestion[] = [];
   activeReplacementId: number = -1;
 
+  constructor();
+  constructor(instance: TrackedContextToken);
+  constructor(instance?: TrackedContextToken) {
+    Object.assign(this, instance);
```

Comment:
Note that this copies the array instances (e.g. `replacements`) rather than deep copy, so changes to `instance.replacements` would be reflected in `this.replacements`...